src/Illuminate/Support/helpers.php

```diff
@@ -508,9 +508,16 @@ function data_set(&$target, $key, $value, $overwrite = true)
      */
     function dd()
     {
-        array_map(function ($x) {
-            (new Dumper)->dump($x);
-        }, func_get_args());
+        $caller = debug_backtrace()[0];
+        $caller = $caller['function'].':'.$caller['line'];
+
+        if (config('app.debug')) {
```

Comment:
What do you think would be a more appropriate toggle? I agree that it's a hidden dependency/coupling, but isn't `app.debug` also how laravel decides when the exception handler should display the exception details vs "something went wrong"? I think it makes sense to  use the same rule.